fvm/src/syscalls/event.rs

```diff
@@ -32,8 +33,21 @@ pub fn emit_event(
     val_off: u32,
     val_len: u32,
 ) -> Result<()> {
-    let raw_event = context.memory.try_slice(event_off, event_len)?;
+    let event_headers = unsafe {
+        const EVENT_SIZE: u32 = std::mem::size_of::<fvm_shared::sys::EventEntry>() as u32;
+        // assert the alignment so we can safely cast from a byte-slice
+        static_assertions::assert_eq_align!(fvm_shared::sys::EventEntry, u8);
+        let size = event_len
+            .checked_mul(EVENT_SIZE)
+            .context("events index out of bounds")
+            .or_illegal_argument()?;
+        let buf = context.memory.try_slice(event_off, size)?;
+        std::slice::from_raw_parts(
+            buf.as_ptr() as *const fvm_shared::sys::EventEntry,
+            event_len as usize,
+        )
```

Comment:
Yeah, it makes sense doing as much condition checking here rather in the kernel.